amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/GracefulShutdownCoordinator.java

```diff
@@ -149,13 +169,17 @@ private boolean workerShutdownWithRemaining(long outstanding, GracefulShutdownCo
         @Override
         public Boolean call() throws Exception {
             GracefulShutdownContext context;
+            boolean recordProcessorsShutdownSuccess;
+            boolean schedulerShutdownSuccess;
             try {
                 context = startWorkerShutdown.call();
+                recordProcessorsShutdownSuccess = context.isRecordProcessorShutdownComplete() || waitForRecordProcessors(context);
```

Comment:
Note: having `context.isRecordProcessorShutdownComplete()` is necessary for the case where a GracefulShutdownContext is created without a null shutdownCompleteLatch and a null notificationCompleteLatch ([ref](https://github.com/awslabs/amazon-kinesis-client/blob/581d713815ef55d8578bd4d51527fe43a43bc72c/amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/Scheduler.java#L800)).